src/bgw/scheduler.c

```diff
@@ -694,6 +706,7 @@ ts_bgw_scheduler_process(int32 run_for_interval_ms,
 	while (quit_time > ts_timer_get_current_timestamp() && !ProcDiePending && !ts_shutdown_bgw)
```

Comment:
This does seem a little backwards to me as well.  I'd have expected you to switch to the necessary context here rather than asserting that the previous loop left you in a particular state.